### DISCUSSION TITLE
Create middleware that run after the controller functions

### DISCUSSION THREAD

**Discussion by marefati110** [strapi/strapi#5924] [Legacy Topics]
**Describe the topic**
Hi, thank you for your help
Is there a way to create a middleware that runs after the controller functions?
Do I want to have a function that applies to all my outputs?

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question that is not really a bug report or a feature request, questions should be directed to the [discussions](https://github.com/strapi/strapi/discussions)

I have converted this to a discussion for you, we try to keep Github clean and only for bug reports and feature requests.

**Comment by lauriejim** [comment]
Hello! When you create your middleware, if you place your code after the `await next()`
The controller / service functions will be called and you will be able to access the response in `ctx.body`

```js
module.exports = strapi => {
  return {
    initialize() {
      strapi.app.use(async (ctx, next) => {

        await next();

        console.log(ctx.body);
      });
    },
  };
};
```
